evaluator/src/main/resources/rules/osadl.rules.kts

```diff
@@ -0,0 +1,110 @@
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+// Use this rule like:
+//
+// $ ort evaluate -i scanner/src/funTest/assets/dummy-expected-output-for-analyzer-result.yml --rules-resource /rules/osadl.rules.kts
+
+import kotlinx.serialization.ExperimentalSerializationApi
+import kotlinx.serialization.json.Json
+import kotlinx.serialization.json.decodeFromStream
+
+enum class Compatibility {
+    INHERENT,
+    YES,
+    NO,
+    UNKNOWN;
+
+    companion object {
+        fun fromString(value: String): Compatibility =
+            when (value) {
+                "Same" -> INHERENT
+                "Yes" -> YES
+                "No" -> NO
+                "Unknown", "Check dependency" -> UNKNOWN
+                else -> throw IllegalArgumentException("Unhandled compatibility '$value' found.")
+            }
+    }
+}
+
+@ExperimentalSerializationApi
+val matrix = javaClass.getResourceAsStream("/rules/osadl-matrix.json").use {
+    Json.Default.decodeFromStream<Map<String, Map<String, String>>>(it)
+}.mapValues { (_, outbounds) -> outbounds.mapValues { (_, value) -> Compatibility.fromString(value) } }
+
+fun PackageRule.LicenseRule.isOsadlMatrixCompliant(outbound: SpdxSingleLicenseExpression) =
+    object : RuleMatcher {
+        private val compatibleValues = enumSetOf(Compatibility.INHERENT, Compatibility.YES)
+
+        override val description = "isOsadlMatrixCompliant($license)"
+
+        @ExperimentalSerializationApi
+        override fun matches(): Boolean {
+            // Be conservative and use the simple license string without the exception string for lookup. Eventually
+            // resulting false-positives are better than an incompatibility to be missed.
+            val inbound = license.simpleLicense()
+            return matrix.getValue(inbound).getValue(outbound.simpleLicense()) in compatibleValues
+        }
+    }
+
+val ruleSet = ruleSet(ortResult, licenseInfoResolver) {
+    val licenseView = LicenseView.CONCLUDED_OR_DECLARED_AND_DETECTED
+
+    // Define a rule that is executed for each dependency.
+    dependencyRule("OSADL compliance matrix") {
+        // Requirements for the rule to trigger a violation.
+        require {
+            -isExcluded()
+        }
+
+        val projectLicenseInfo = licenseInfoResolver.resolveLicenseInfo(project.id).filter(licenseView)
+        val outboundLicenses = projectLicenseInfo.licenses.map { it.license }
+
+        // Define a rule that is executed for each license of the dependency.
+        licenseRule("OSADL compliance matrix", licenseView) {
+            outboundLicenses.forEach { outboundLicense ->
+                if (isOsadlMatrixCompliant(outboundLicense).matches()) return@forEach
+
+                val inboundLicenseString = if (license.exception() != null) {
+                    "'${license.simpleLicense()}' (simplified from '$license')"
+                } else {
+                    "'$license'"
+                }
```

Comment:
Turning the license into this license string should be factored out into a function.